src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java

```diff
@@ -185,19 +176,11 @@ private Object readResolve() throws IOException {
     }
 
     try {
-      this.mesosApi =
-          MesosApi.getInstance(
-              this.master,
-              this.jenkinsURL,
-              this.agentUser,
-              this.frameworkName,
-              this.frameworkId,
-              this.role,
-              this.sslCert,
-              this.dcosAuthorization);
-      logger.info("Initialized Mesos API object after deserialization.");
+      MesosApi.getInstance(this);
```

Comment:
This seems ill adviced, same reason why it was ill adviced to do this in the constructor.

Reply:
Thanks for the review. This line is not required.